Rebuilt rgbeffects file

MrPaulAR opened the pull request:
I would suggest using this instead of the default rgbeffects file because when working in the sequencer there is *much* less lag.  Also, there are quite a few groups that have been created which allow more effects such as the single line.  This better aligns with how xLights was traditionally used on the house model.

The only sacrifice in this layout is that it breaks the sample show.  If needed I can fix that too but a quick import will address that.

MrPaulAR committed:
Rebuilt rgbeffects file

I would suggest using this instead of the default rgbeffects file because when working in the sequencer there is *much* less lag.  Also, there are quite a few groups that have been created which allow more effects such as the single line.  This better aligns with how xLights was traditionally used on the house model.

The only sacrifice in this layout is that it breaks the sample show.  If needed I can fix that too but a quick import will address that.

JVB-Tesla commented:
![model img](https://user-images.githubusercontent.com/96159903/148465108-6286bc74-9fc9-4314-8f73-6e60c63e3bcf.png)

@MrPaulAR , this really speeds things up! Much appreciated.

It looks like there are some xLights channel mapping errors. If you can address these, we can merge. Would also appreciate you updating the example project.

MrPaulAR committed:
Add all group

MrPaulAR committed:
Imported sample sequence

MrPaulAR committed:
Mapping file for importing old sequences

MrPaulAR commented:
I've went through the rgbeffects file and I'm not seeing any missing channels but I did add a new group for all the effects.  If you see something specific please let me know and I will fix.

I also went through and imported the sample show that was provided in readme.md.  In this I observed that one of your rear door handles was misnamed (rear rear door handle).  I took the liberty of uploading this xsq & wav file into a Sequences/Sample folder.  This will help keep things organized IMHO as all sequences will be under this folder structure.

```bash
├── Sequences
│   ├── Sample
│   │   ├── lightshow.wav # Audio file
│   │   ├── lightshow.xsq # xLights sequence
│   │   ├── lightshow.fseq # Sequence that is excluded from github and generated by xLights
│   │   ├── mapping.xmap # Mapping file ... see below
│   ├── User Sequence 1
│   │   ├── lightshow.wav # Audio file
│   │   ├── lightshow.xsq # xLights sequence
│   │   ├── lightshow.fseq # Sequence that is excluded from github and generated by xLights
│   ├── User Sequence 2
│   │   ├── lightshow.wav # Audio file
│   │   ├── lightshow.xsq # xLights sequence
│   │   ├── lightshow.fseq # Sequence that is excluded from github and generated by xLights
│   xlights_keybindings.xml # Not important in this context and not part of repo
│   xlights_networks.xml # Not important in this context and not part of repo
│   xlights_rgbeffects.xml
```

Lastly I included the mapping.xmap file in the Sequences/Sample folder.  This can be used by anyone to import their previous version of xsq into this method.  It's the same one I used to create the xsq in the Sample folder.  As long as someone didn't create groups or something in the previous method it's a 30 second task to convert.

adamcoulombe commented:
![Screenshot_31](https://user-images.githubusercontent.com/469038/148557926-ae47b67e-d952-4455-8ba4-0a3ec0a80b72.png)
These were the errors I got when I used the rgbeffects file. These might be the same channel mapping errors @JVB-Tesla is referring to.

@MrPaulAR I think you'll want to replace the rgbeffects file in the provided zip file too so that you dont have to manually copy it over.

One other thing I noticed that I'm not sure has to do with the new setup, when I try to reposition my house preview, it doesnt refresh or isnt responsive, is this just me?

MrPaulAR committed:
Added networks file to account for auto layout errors

MrPaulAR committed:
Properly defined show folder layout based on xLights best practices

MrPaulAR commented:
I fixed the layout errors by adding the xlights_networks.xml file.  The error was due to the layout referencing a null controller by name for channel assignments as this is much quicker than clicking on a bunch of models individually.  

Regarding the zip.  I don't see any reason for it.  As a matter of fact I just pushed a commit where I properly setup the xLights show folder based on xLights best practices based on their zoom room.

MrPaulAR commented:
Additional comment on the zip.  All but a couple images are inside that zip are just text files.  IMHO we should let git version these.

adamcoulombe commented:
> Additional comment on the zip. All but a couple images are inside that zip are just text files. IMHO we should let git version these.

I tend to agree. I think they wanted to make it easy for those who are unfamiliar to download just the show folder. 

I see you've removed the zip from your latest commit which might raise a separate issue; the readme still explains specifically to download that zip file:
https://github.com/MrPaulAR/light-show/blob/17cab69746b6173677d51616332d8cd5fb66492c/README.md?plain=1#L48

MrPaulAR committed:
Updated readme to reflect the sequencer changes

MrPaulAR commented:
I don't think it is any easier to download a separate zip and extract it than to just download this one repo (clone or zip) but that is for the Tesla team to decide.

Good catch on the README.  

Since my proposal is to have everything for xLights self contained without a zip I'm going to add a commit for that readme as well.

My new worry is that this has grown well past my original scope even though this is all directly related.

adamcoulombe commented:
Yeah, agree. That was the only reason I kept the zip intact in my PR, was to keep the scope limited.

Anyway, I think you actually missed the reference in question though:

https://github.com/MrPaulAR/light-show/blob/e1e8b3d57f67b8cfee563b59d62ebfd30371c575/README.md?plain=1#L48

> 2. Download and unzip [tesla_xlights_show_folder.zip](xlights/tesla_xlights_show_folder.zip?raw=true), which is the Tesla xLights bare project directory.

MrPaulAR commented:
That appears to be in my current PR

![image](https://user-images.githubusercontent.com/3765696/148590583-33d40e18-26da-4d16-b299-cd052406bfbf.png)

adamcoulombe commented:
That is line 77, the line in question is line 48 where it talks about show folder (not the example sequence)

MrPaulAR committed:
Additional verbage related to show folder tweaks

MrPaulAR commented:
I've added some additional verbiage to account for this.  I've also removed a reference to the Model S xmodel since it's no longer one model but an entire layout.

adamcoulombe committed:
Fix Model S obj not showing

Make path to Model S OBJ relative so that Xlights can find it.

I also restored the assets for the model S obj back in the show folder root since it is not exclusive to a sequence

adamcoulombe commented:
Cool, nice work Paul.

I did test out your latest commit and found an issue with the model S no longer displaying, I sent note to you about this in a [pull request for your patch-1 branch ](https://github.com/MrPaulAR/light-show/pull/1)

MrPaulAR committed:
Merge pull request #1 from adamcoulombe/pr_40

Fix Model S obj not showing

MrPaulAR commented:
Merged your pull.  Thanks alot for doing the pull request.  I'm currently just using web interface & github.dev for these as my stuff is currently in a private repo due to a bunch of sequences being there too.

simon2022 commented:
Thanks for your work, it's so much more fun to create lightshows without the lags of the program.
It's also much better with the groups, thank you!

@MrPaulAR could it be that the "Right Side Repeater" is not present in the 3D model?
![lsr](https://user-images.githubusercontent.com/59486823/163158363-77b27003-c548-4c4a-98d7-35b10b4bda69.JPG)

Many greetings

spaceXrace commented:
Wow! Thank you so much. With nearly 18 lightshows programmed, I have been incredibly annoyed by the poor performance of XLights. This makes programming the shows so much faster and so much more fun. 
I made people aware of your patches on r/TeslaLightShow, I hope you're ok with this!

Obereigwe commented:
SuiCloudMining is a premier crypto asset mining company that offers miners the opportunity to accrue a steady stream of passive income through crypto mining. With a 4.5 APY mining rate, our miners can enjoy a steady stream of income that’s independent of market fluctuations. We provide our miners with the best in class mining hardware and software which ensures that our miners are able to mine efficiently and with maximum profitability. Our team of experts also provides our clients with the best in class customer support so that they can get the most out of their mining experience.
Suicloudmining

spaceXrace commented:
The performance improvements were implemented with #58
